gallery_dl/extractor/tiktok.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.tiktok.com/"""
+
+from .common import Extractor, Message
+from .. import exception, text, util
+import re
+
+BASE_PATTERN = r"(?:https?://)?(?:www\.)?tiktok\.com"
+USER_PATTERN = BASE_PATTERN + r"/+@([\w.]{0,23}\w)(?:/\S*)?/*"
+POST_PATTERN = BASE_PATTERN + r"/+@(?:[\w.]{0,23}\w)(?:/\S*)?/+(?:[pP][hH][oO][tT][oO]|[vV][iI][dD][eE][oO])/+(?:[0-9]+)/*"
+VM_POST_PATTERN = r"(?:https?://)?(?:vm\.)?tiktok\.com/+.*/*"
+INSENSITIVE_PHOTO = re.compile(re.escape("/photo/"), re.IGNORECASE)
+
+
+class TikTokExtractor(Extractor):
+    """Base class for TikTok extractors"""
+
+    category = "tiktok"
+    directory_fmt = ("{category}", "{user}")
+    filename_fmt = "{id}_{index}_{img_id}.{extension}"
```

Comment:
Note to self: the filename should probably follow the default yt-dlp `outtmpl` for TikTok (`{title} [{video_id}] [{index}].[{extension}]`. Don't forget to sanitize the title of invalid characters and limit its length.